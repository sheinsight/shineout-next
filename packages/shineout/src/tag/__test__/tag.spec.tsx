import { render, cleanup, fireEvent, waitFor } from '@testing-library/react';
import Tag from '..';
import {
  snapshotTest,
  classTest,
  textContentTest,
  classContentTest,
  styleTest,
  baseTest,
  childrenTest,
  inputValueTest,
  displayTest,
} from '../../tests/utils';
import mountTest from '../../tests/mountTest';
import structureTest, {
  classLengthTest,
  structureTestInContainer,
} from '../../tests/structureTest';
import TagBase from '../__example__/s-001-base';
import TagSize from '../__example__/s-002-size';
import TagMode from '../__example__/s-003-mode';
import TagColor from '../__example__/s-004-color';
import TagDynamic from '../__example__/s-005-dynamic';
import TagEditable from '../__example__/s-006-editable';
import TagShape from '../__example__/s-007-shape';
import { useState } from 'react';

const SO_PREFIX = 'tag';
const tagClassName = `.${SO_PREFIX}-tag-0-2-1`;
const tagSmallClassName = `${SO_PREFIX}-small-0-2-5`;
const tagBrightClassName = `${SO_PREFIX}-bright-0-2-23`;
const tagDefaultClassName = `${SO_PREFIX}-default-0-2-10`;
const tagInlineClassName = `.${SO_PREFIX}-inline-0-2-3`;
const tagCloseClassName = `.${SO_PREFIX}-closeIcon-0-2-7`;
const tagDisabledClassName = `${SO_PREFIX}-disabled-0-2-6`;
const tagRoundedClassName = `${SO_PREFIX}-rounded-0-2-26`;
const tagInputClassName = `.${SO_PREFIX}-input-0-2-2`;

const TagColorArray = ['default', 'info', 'danger', 'warning', 'success'];
const TagSizeArray = ['Small', 'Default', 'Large'];
const TageModeArray = ['bright', 'fill', 'outline', 'brightOutline'];
const TagColorMoreArray = [
  'tangerine',
  'magenta',
  'purple',
  'indigo',
  'cyan',
  'neon',
  'lemon',
  'orange',
];
const closeAttributes = [
  {
    attribute: tagInlineClassName,
    num: 1,
  },
  {
    attribute: tagCloseClassName,
    num: 1,
  },
  {
    attribute: 'svg',
    num: 1,
  },
];
const backgroundColor = 'rgb(0, 0, 0)';

afterEach(cleanup);
describe('Tag[Base]', () => {
  mountTest(Tag);
  displayTest(Tag, 'ShineoutTag');
  snapshotTest(<TagBase />);
  snapshotTest(<TagSize />, 'about size');
  snapshotTest(<TagMode />, 'about mode');
  snapshotTest(<TagColor />, 'about color');
  test('should render when set different color', () => {
    const { container } = render(<TagBase />);
    const tags = container.querySelectorAll(tagClassName)!;
    classLengthTest(container, tagClassName, TagColorArray.length);
    tags.forEach((tag, index) => {
      classTest(tag, tagSmallClassName);
      classTest(tag, tagBrightClassName);
      classContentTest(tag, TagColorArray[index]);
      textContentTest(tag, TagColorArray[index]);
    });
  });
  test('should render when set different size', () => {
    const { container } = render(<TagSize />);
    const tags = container.querySelectorAll(tagClassName)!;
    classLengthTest(container, tagClassName, TagSizeArray.length);
    tags.forEach((tag, index) => {
      classTest(tag, tagDefaultClassName);
      classTest(tag, tagBrightClassName);
      textContentTest(tag, TagSizeArray[index]);
      if (tag.textContent === TagSizeArray[1]) return;
      classContentTest(tag, TagSizeArray[index].toLowerCase());
    });
  });
  test('should render when set different mode', () => {
    const { container } = render(<TagMode />);
    const tags = container.querySelectorAll(tagClassName)!;
    classLengthTest(container, tagClassName, TagColorArray.length * TageModeArray.length);
    tags.forEach((tag, index) => {
      classContentTest(tag, TageModeArray[Math.floor(index / TagColorArray.length)]);
      classContentTest(tag, TagColorArray[index % TagColorArray.length]);
    });
  });
  test('should render when set different more color', () => {
    const { container } = render(<TagColor />);
    const tags = container.querySelectorAll(tagClassName)!;
    classLengthTest(container, tagClassName, TagColorMoreArray.length * TageModeArray.length);
    tags.forEach((tag, index) => {
      classContentTest(tag, TageModeArray[Math.floor(index / TagColorMoreArray.length)]);
      classContentTest(tag, TagColorMoreArray[index % TagColorMoreArray.length]);
    });
  });
  baseTest(Tag, tagClassName);
});
describe('Tag[onClose]', () => {
  structureTest(<Tag onClose>Hello</Tag>, closeAttributes);
  test('should render when set onClose is boolean', () => {
    const closeAttributesAfter = closeAttributes.map((v) => ({ attribute: v.attribute, num: 0 }));
    const { container } = render(<Tag onClose>Hello</Tag>);
    const tag = container.querySelector(tagClassName)!;
    fireEvent.click(tag.querySelector(tagCloseClassName)!);
    structureTestInContainer(container, closeAttributesAfter);
  });
  test('should not render close default', () => {
    const { container } = render(<Tag>Hello</Tag>);
    classLengthTest(container, tagCloseClassName, 0);
  });
  test('should render when set onClose is function', () => {
    const closeAttributesAfter = closeAttributes.map((v) => ({ attribute: v.attribute, num: 0 }));
    const closeFn = jest.fn();
    const { container } = render(<Tag onClose={closeFn}>Hello</Tag>);
    structureTestInContainer(container, closeAttributes);
    fireEvent.click(container.querySelector(tagCloseClassName)!);
    structureTestInContainer(container, closeAttributesAfter);
    expect(closeFn.mock.calls.length).toBe(1);
  });
  test('should render when click with defaultPrevented', () => {
    const closeAttributesAfter = closeAttributes.map((v) => ({ attribute: v.attribute, num: 0 }));
    const closeFn = jest.fn();
    const { container } = render(<Tag onClose={closeFn}>Hello</Tag>);
    fireEvent.click(container.querySelector(tagCloseClassName)!, { defaultPrevented: true });
    structureTestInContainer(container, closeAttributesAfter);
    expect(closeFn.mock.calls.length).toBe(1);
  });
  test('should render when set onClose is promise', async () => {
    jest.useFakeTimers();
    const closeFn = jest.fn();
    const { container } = render(
      <Tag
        onClose={() =>
          new Promise((resolve) => {
            setTimeout(() => {
              closeFn();
              resolve(true);
            }, 3000);
          })
        }
      >
        Hello
      </Tag>,
    );
    structureTestInContainer(container, closeAttributes);
    fireEvent.click(container.querySelector(tagCloseClassName)!);
    // TODO: 还没有spin
    jest.runAllTimers();
    await waitFor(() => {
      expect(closeFn.mock.calls.length).toBe(1);
    });
  });
  test('should render when set onClose is promise reject', async () => {
    jest.useFakeTimers();
    const closeFn = jest.fn();
    const { container } = render(
      <Tag
        onClose={() =>
          new Promise((resolve, reject) => {
            setTimeout(() => {
              closeFn();
              reject();
            }, 3000);
          })
        }
      >
        Hello
      </Tag>,
    );
    structureTestInContainer(container, closeAttributes);
    fireEvent.click(container.querySelector(tagCloseClassName)!);
    // TODO: 还没有spin
    jest.runAllTimers();
    await waitFor(() => {
      expect(closeFn.mock.calls.length).toBe(1);
    });
  });
});
describe('Tag[Disabled]', () => {
  test('should render when set disabled', () => {
    const { container } = render(<Tag disabled>Hello</Tag>);
    classTest(container.querySelector(tagClassName)!, tagDisabledClassName);
  });
  test('should not close when set disabled', () => {
    const { container } = render(
      <Tag disabled onClose>
        Hello
      </Tag>,
    );
    fireEvent.click(container.querySelector(tagCloseClassName)!);
    classLengthTest(container, tagClassName, 1);
  });
  //TODO: 可编辑disabled控制可以放在onCompleted中
});
describe('Tag[BackgroundColor]', () => {
  test('should render when set background color', () => {
    const { container } = render(<Tag backgroundColor={backgroundColor}>Hello</Tag>);
    styleTest(container.querySelector(tagClassName)!, 'background-color: rgb(0, 0, 0);');
  });
  test('should render when set background color and style', () => {
    const { container } = render(
      <Tag backgroundColor={backgroundColor} style={{ backgroundColor: 'red' }}>
        Hello
      </Tag>,
    );
    styleTest(container.querySelector(tagClassName)!, 'background-color: rgb(0, 0, 0);');
  });
});
describe('Tag[children]', () => {
  childrenTest(Tag, tagClassName);
});
describe('Tag[onClick]', () => {
  test('should onClick', () => {
    const clickFn = jest.fn();
    const { container } = render(<Tag onClick={clickFn} />);
    fireEvent.click(container.querySelector(tagClassName)!);
    expect(clickFn.mock.calls.length).toBe(1);
  });
});
describe('Tag[Shape]', () => {
  snapshotTest(<TagShape />, 'about shape');
  test('should render when set shape', () => {
    const { container } = render(<TagShape />);
    const tags = container.querySelectorAll(tagClassName);
    tags.forEach((tag) => {
      if (tag.textContent === 'Default') return;
      classTest(tag, tagRoundedClassName);
    });
  });
});
describe('Tag[Input]', () => {
  snapshotTest(<TagEditable />, 'about input');
  snapshotTest(<TagDynamic />, 'about dynamic');
  test('should render when set onCompleted', () => {
    const { container } = render(<TagEditable />);
    const tag = container.querySelector(tagClassName);
    const inline = tag?.querySelector(tagInlineClassName) as Element;
    textContentTest(inline, 'abc');
    classLengthTest(container, 'input', 0);
    fireEvent.click(tag!);
    classLengthTest(container, 'input', 1);
    const input = container.querySelector('input')!;
    inputValueTest(input, 'abc');
    fireEvent.change(input, {
      target: { value: 'test' },
    });
    inputValueTest(input, 'test');
    fireEvent.blur(input);
    classLengthTest(container, 'input', 0);
    textContentTest(inline, 'abc');
  });
  test('should render when set tag input', () => {
    const focusFn = jest.fn();
    const keyUpFn = jest.fn();
    const TestDemo = () => {
      const [value, setValue] = useState<string | undefined>('abc');
      return (
        <Tag.Input
          onChange={(e) => {
            setValue(e);
          }}
          value={value}
          onFocus={focusFn}
          onKeyUp={keyUpFn}
        />
      );
    };
    const { container } = render(<TestDemo />);
    const input = container.querySelector(tagInputClassName)!;
    const inputContent = input.querySelector('input')!;
    expect(input).toBeTruthy();
    classLengthTest(input, 'input', 1);
    inputValueTest(inputContent, 'abc');
    fireEvent.focus(inputContent);
    expect(focusFn.mock.calls.length).toBe(2);
    fireEvent.keyUp(inputContent, { key: 'Enter' });
    expect(keyUpFn.mock.calls.length).toBe(1);
    fireEvent.change(inputContent, {
      target: { value: 'test' },
    });
    inputValueTest(inputContent, 'test');
  });
  test('should render dynamic', () => {
    const { container } = render(<TagDynamic />);
    classLengthTest(container, tagClassName, 4);
    const tags = container.querySelectorAll(tagClassName)!;
    textContentTest(tags[3], '+ New Tag');
    classLengthTest(container, 'input', 0);
    fireEvent.click(tags[0].querySelector(tagCloseClassName)!);
    classLengthTest(container, tagClassName, 3);
    fireEvent.click(tags[3]);
    classLengthTest(container, 'input', 1);
    fireEvent.change(container.querySelector('input')!, {
      target: { value: 'test' },
    });
    fireEvent.blur(container.querySelector('input')!);
    classLengthTest(container, tagClassName, 4);
    const tagsNew = container.querySelectorAll(tagClassName)!;
    textContentTest(tagsNew[2], 'test');
  });
  test('should render when set value is fixed value', () => {
    const changeFn = jest.fn();
    const { container } = render(<Tag.Input onChange={changeFn} value='test' />);
    fireEvent.click(container.querySelector('input')!);
    expect(changeFn.mock.calls.length).toBe(0);
  });
});
describe('Tag[onKeyUp/onEnterPress]', () => {
  test('should onKeyUp', () => {
    const keyUpFn = jest.fn();
    const completedFn = jest.fn();
    const { container } = render(
      <Tag onCompleted={completedFn} onKeyUp={keyUpFn}>
        test
      </Tag>,
    );
    const tag = container.querySelector(tagClassName)!;
    fireEvent.click(tag);
    classLengthTest(container.querySelector(tagInputClassName)!, 'input', 1);
    fireEvent.keyUp(container.querySelector('input')!, { keyCode: 13, target: { value: 'Hello' } });
    expect(keyUpFn.mock.calls.length).toBe(1);
  });
  test('should onEnterPress', () => {
    const enterPressFn = jest.fn();
    const completedFn = jest.fn();
    const { container } = render(
      <Tag onCompleted={completedFn} onEnterPress={enterPressFn}>
        test
      </Tag>,
    );
    const tag = container.querySelector(tagClassName)!;
    fireEvent.click(tag);
    classLengthTest(container.querySelector(tagInputClassName)!, 'input', 1);
    fireEvent.keyUp(container.querySelector('input')!, { keyCode: 13, target: { value: 'Hello' } });
    expect(enterPressFn.mock.calls.length).toBe(1);
  });
});
