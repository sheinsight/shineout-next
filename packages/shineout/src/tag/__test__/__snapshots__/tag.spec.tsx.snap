// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag[Base] should render correctly  1`] = `
<div>
  <div
    class="tag-tag-0-2-1 tag-small-0-2-5 tag-default-0-2-10 tag-bright-0-2-23"
  >
    <span>
      default
    </span>
  </div>
  <div
    class="tag-tag-0-2-1 tag-small-0-2-5 tag-info-0-2-9 tag-bright-0-2-23"
  >
    <span>
      info
    </span>
  </div>
  <div
    class="tag-tag-0-2-1 tag-small-0-2-5 tag-danger-0-2-13 tag-bright-0-2-23"
  >
    <span>
      danger
    </span>
  </div>
  <div
    class="tag-tag-0-2-1 tag-small-0-2-5 tag-warning-0-2-12 tag-bright-0-2-23"
  >
    <span>
      warning
    </span>
  </div>
  <div
    class="tag-tag-0-2-1 tag-small-0-2-5 tag-success-0-2-11 tag-bright-0-2-23"
  >
    <span>
      success
    </span>
  </div>
</div>
`;

exports[`Tag[Base] should render correctly about color 1`] = `
<div>
  <div
    style="margin-bottom: 24px;"
  >
    <div
      class="tag-tag-0-2-1 tag-tangerine-0-2-21 tag-bright-0-2-23"
    >
      <span>
        Tag
      </span>
    </div>
    <div
      class="tag-tag-0-2-1 tag-magenta-0-2-15 tag-bright-0-2-23"
    >
      <span>
        Tag
      </span>
    </div>
    <div
      class="tag-tag-0-2-1 tag-purple-0-2-16 tag-bright-0-2-23"
    >
      <span>
        Tag
      </span>
    </div>
    <div
      class="tag-tag-0-2-1 tag-indigo-0-2-17 tag-bright-0-2-23"
    >
      <span>
        Tag
      </span>
    </div>
    <div
      class="tag-tag-0-2-1 tag-cyan-0-2-18 tag-bright-0-2-23"
    >
      <span>
        Tag
      </span>
    </div>
    <div
      class="tag-tag-0-2-1 tag-neon-0-2-19 tag-bright-0-2-23"
    >
      <span>
        Tag
      </span>
    </div>
    <div
      class="tag-tag-0-2-1 tag-lemon-0-2-20 tag-bright-0-2-23"
    >
      <span>
        Tag
      </span>
    </div>
    <div
      class="tag-tag-0-2-1 tag-orange-0-2-14 tag-bright-0-2-23"
    >
      <span>
        Tag
      </span>
    </div>
  </div>
  <div
    style="margin-bottom: 24px;"
  >
    <div
      class="tag-tag-0-2-1 tag-tangerine-0-2-21 tag-fill-0-2-22"
    >
      <span>
        Tag
      </span>
    </div>
    <div
      class="tag-tag-0-2-1 tag-magenta-0-2-15 tag-fill-0-2-22"
    >
      <span>
        Tag
      </span>
    </div>
    <div
      class="tag-tag-0-2-1 tag-purple-0-2-16 tag-fill-0-2-22"
    >
      <span>
        Tag
      </span>
    </div>
    <div
      class="tag-tag-0-2-1 tag-indigo-0-2-17 tag-fill-0-2-22"
    >
      <span>
        Tag
      </span>
    </div>
    <div
      class="tag-tag-0-2-1 tag-cyan-0-2-18 tag-fill-0-2-22"
    >
      <span>
        Tag
      </span>
    </div>
    <div
      class="tag-tag-0-2-1 tag-neon-0-2-19 tag-fill-0-2-22"
    >
      <span>
        Tag
      </span>
    </div>
    <div
      class="tag-tag-0-2-1 tag-lemon-0-2-20 tag-fill-0-2-22"
    >
      <span>
        Tag
      </span>
    </div>
    <div
      class="tag-tag-0-2-1 tag-orange-0-2-14 tag-fill-0-2-22"
    >
      <span>
        Tag
      </span>
    </div>
  </div>
  <div
    style="margin-bottom: 24px;"
  >
    <div
      class="tag-tag-0-2-1 tag-tangerine-0-2-21 tag-outline-0-2-24"
    >
      <span>
        Tag
      </span>
    </div>
    <div
      class="tag-tag-0-2-1 tag-magenta-0-2-15 tag-outline-0-2-24"
    >
      <span>
        Tag
      </span>
    </div>
    <div
      class="tag-tag-0-2-1 tag-purple-0-2-16 tag-outline-0-2-24"
    >
      <span>
        Tag
      </span>
    </div>
    <div
      class="tag-tag-0-2-1 tag-indigo-0-2-17 tag-outline-0-2-24"
    >
      <span>
        Tag
      </span>
    </div>
    <div
      class="tag-tag-0-2-1 tag-cyan-0-2-18 tag-outline-0-2-24"
    >
      <span>
        Tag
      </span>
    </div>
    <div
      class="tag-tag-0-2-1 tag-neon-0-2-19 tag-outline-0-2-24"
    >
      <span>
        Tag
      </span>
    </div>
    <div
      class="tag-tag-0-2-1 tag-lemon-0-2-20 tag-outline-0-2-24"
    >
      <span>
        Tag
      </span>
    </div>
    <div
      class="tag-tag-0-2-1 tag-orange-0-2-14 tag-outline-0-2-24"
    >
      <span>
        Tag
      </span>
    </div>
  </div>
  <div
    style="margin-bottom: 24px;"
  >
    <div
      class="tag-tag-0-2-1 tag-tangerine-0-2-21 tag-brightOutline-0-2-25"
    >
      <span>
        Tag
      </span>
    </div>
    <div
      class="tag-tag-0-2-1 tag-magenta-0-2-15 tag-brightOutline-0-2-25"
    >
      <span>
        Tag
      </span>
    </div>
    <div
      class="tag-tag-0-2-1 tag-purple-0-2-16 tag-brightOutline-0-2-25"
    >
      <span>
        Tag
      </span>
    </div>
    <div
      class="tag-tag-0-2-1 tag-indigo-0-2-17 tag-brightOutline-0-2-25"
    >
      <span>
        Tag
      </span>
    </div>
    <div
      class="tag-tag-0-2-1 tag-cyan-0-2-18 tag-brightOutline-0-2-25"
    >
      <span>
        Tag
      </span>
    </div>
    <div
      class="tag-tag-0-2-1 tag-neon-0-2-19 tag-brightOutline-0-2-25"
    >
      <span>
        Tag
      </span>
    </div>
    <div
      class="tag-tag-0-2-1 tag-lemon-0-2-20 tag-brightOutline-0-2-25"
    >
      <span>
        Tag
      </span>
    </div>
    <div
      class="tag-tag-0-2-1 tag-orange-0-2-14 tag-brightOutline-0-2-25"
    >
      <span>
        Tag
      </span>
    </div>
  </div>
</div>
`;

exports[`Tag[Base] should render correctly about mode 1`] = `
<div>
  <div
    style="margin-bottom: 8px;"
  >
    <div
      class="tag-tag-0-2-1 tag-default-0-2-10 tag-bright-0-2-23"
    >
      <span>
        default
      </span>
    </div>
    <div
      class="tag-tag-0-2-1 tag-info-0-2-9 tag-bright-0-2-23"
    >
      <span>
        info
      </span>
    </div>
    <div
      class="tag-tag-0-2-1 tag-danger-0-2-13 tag-bright-0-2-23"
    >
      <span>
        danger
      </span>
    </div>
    <div
      class="tag-tag-0-2-1 tag-warning-0-2-12 tag-bright-0-2-23"
    >
      <span>
        warning
      </span>
    </div>
    <div
      class="tag-tag-0-2-1 tag-success-0-2-11 tag-bright-0-2-23"
    >
      <span>
        success
      </span>
    </div>
  </div>
  <div
    style="margin-bottom: 8px;"
  >
    <div
      class="tag-tag-0-2-1 tag-default-0-2-10 tag-fill-0-2-22"
    >
      <span>
        default
      </span>
    </div>
    <div
      class="tag-tag-0-2-1 tag-info-0-2-9 tag-fill-0-2-22"
    >
      <span>
        info
      </span>
    </div>
    <div
      class="tag-tag-0-2-1 tag-danger-0-2-13 tag-fill-0-2-22"
    >
      <span>
        danger
      </span>
    </div>
    <div
      class="tag-tag-0-2-1 tag-warning-0-2-12 tag-fill-0-2-22"
    >
      <span>
        warning
      </span>
    </div>
    <div
      class="tag-tag-0-2-1 tag-success-0-2-11 tag-fill-0-2-22"
    >
      <span>
        success
      </span>
    </div>
  </div>
  <div
    style="margin-bottom: 8px;"
  >
    <div
      class="tag-tag-0-2-1 tag-default-0-2-10 tag-outline-0-2-24"
    >
      <span>
        default
      </span>
    </div>
    <div
      class="tag-tag-0-2-1 tag-info-0-2-9 tag-outline-0-2-24"
    >
      <span>
        info
      </span>
    </div>
    <div
      class="tag-tag-0-2-1 tag-danger-0-2-13 tag-outline-0-2-24"
    >
      <span>
        danger
      </span>
    </div>
    <div
      class="tag-tag-0-2-1 tag-warning-0-2-12 tag-outline-0-2-24"
    >
      <span>
        warning
      </span>
    </div>
    <div
      class="tag-tag-0-2-1 tag-success-0-2-11 tag-outline-0-2-24"
    >
      <span>
        success
      </span>
    </div>
  </div>
  <div
    style="margin-bottom: 8px;"
  >
    <div
      class="tag-tag-0-2-1 tag-default-0-2-10 tag-brightOutline-0-2-25"
    >
      <span>
        default
      </span>
    </div>
    <div
      class="tag-tag-0-2-1 tag-info-0-2-9 tag-brightOutline-0-2-25"
    >
      <span>
        info
      </span>
    </div>
    <div
      class="tag-tag-0-2-1 tag-danger-0-2-13 tag-brightOutline-0-2-25"
    >
      <span>
        danger
      </span>
    </div>
    <div
      class="tag-tag-0-2-1 tag-warning-0-2-12 tag-brightOutline-0-2-25"
    >
      <span>
        warning
      </span>
    </div>
    <div
      class="tag-tag-0-2-1 tag-success-0-2-11 tag-brightOutline-0-2-25"
    >
      <span>
        success
      </span>
    </div>
  </div>
</div>
`;

exports[`Tag[Base] should render correctly about size 1`] = `
<div>
  <div
    class="tag-tag-0-2-1 tag-small-0-2-5 tag-default-0-2-10 tag-bright-0-2-23"
  >
    <span>
      Small
    </span>
  </div>
  <div
    class="tag-tag-0-2-1 tag-default-0-2-10 tag-bright-0-2-23"
  >
    <span>
      Default
    </span>
  </div>
  <div
    class="tag-tag-0-2-1 tag-large-0-2-4 tag-default-0-2-10 tag-bright-0-2-23"
  >
    <span>
      Large
    </span>
  </div>
</div>
`;

exports[`Tag[Input] should render correctly about dynamic 1`] = `
<div>
  <div
    class="tag-tag-0-2-1 tag-default-0-2-10 tag-bright-0-2-23"
  >
    <div
      class="tag-inline-0-2-3"
    >
      <span>
        Tag 1
      </span>
    </div>
    <div
      class="tag-closeIcon-0-2-7"
    >
      <span
        class="tag-closeIconWrapper-0-2-8"
      >
        <svg
          height="10"
          viewBox="0 0 10 10"
          width="10"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M1.92204 1.33738L1.96129 1.37204L5 4.41042L8.03871 1.37204C8.20142 1.20932 8.46524 1.20932 8.62796 1.37204C8.77816 1.52224 8.78972 1.75859 8.66262 1.92204L8.62796 1.96129L5.58958 5L8.62796 8.03871C8.79068 8.20142 8.79068 8.46524 8.62796 8.62796C8.47776 8.77816 8.24141 8.78972 8.07796 8.66262L8.03871 8.62796L5 5.58958L1.96129 8.62796C1.79858 8.79068 1.53476 8.79068 1.37204 8.62796C1.22184 8.47776 1.21028 8.24141 1.33738 8.07796L1.37204 8.03871L4.41042 5L1.37204 1.96129C1.20932 1.79858 1.20932 1.53476 1.37204 1.37204C1.52224 1.22184 1.75859 1.21028 1.92204 1.33738Z"
          />
        </svg>
      </span>
    </div>
  </div>
  <div
    class="tag-tag-0-2-1 tag-default-0-2-10 tag-bright-0-2-23"
  >
    <div
      class="tag-inline-0-2-3"
    >
      <span>
        Tag 2
      </span>
    </div>
    <div
      class="tag-closeIcon-0-2-7"
    >
      <span
        class="tag-closeIconWrapper-0-2-8"
      >
        <svg
          height="10"
          viewBox="0 0 10 10"
          width="10"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M1.92204 1.33738L1.96129 1.37204L5 4.41042L8.03871 1.37204C8.20142 1.20932 8.46524 1.20932 8.62796 1.37204C8.77816 1.52224 8.78972 1.75859 8.66262 1.92204L8.62796 1.96129L5.58958 5L8.62796 8.03871C8.79068 8.20142 8.79068 8.46524 8.62796 8.62796C8.47776 8.77816 8.24141 8.78972 8.07796 8.66262L8.03871 8.62796L5 5.58958L1.96129 8.62796C1.79858 8.79068 1.53476 8.79068 1.37204 8.62796C1.22184 8.47776 1.21028 8.24141 1.33738 8.07796L1.37204 8.03871L4.41042 5L1.37204 1.96129C1.20932 1.79858 1.20932 1.53476 1.37204 1.37204C1.52224 1.22184 1.75859 1.21028 1.92204 1.33738Z"
          />
        </svg>
      </span>
    </div>
  </div>
  <div
    class="tag-tag-0-2-1 tag-default-0-2-10 tag-bright-0-2-23"
  >
    <div
      class="tag-inline-0-2-3"
    >
      <span>
        Tag 3
      </span>
    </div>
    <div
      class="tag-closeIcon-0-2-7"
    >
      <span
        class="tag-closeIconWrapper-0-2-8"
      >
        <svg
          height="10"
          viewBox="0 0 10 10"
          width="10"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M1.92204 1.33738L1.96129 1.37204L5 4.41042L8.03871 1.37204C8.20142 1.20932 8.46524 1.20932 8.62796 1.37204C8.77816 1.52224 8.78972 1.75859 8.66262 1.92204L8.62796 1.96129L5.58958 5L8.62796 8.03871C8.79068 8.20142 8.79068 8.46524 8.62796 8.62796C8.47776 8.77816 8.24141 8.78972 8.07796 8.66262L8.03871 8.62796L5 5.58958L1.96129 8.62796C1.79858 8.79068 1.53476 8.79068 1.37204 8.62796C1.22184 8.47776 1.21028 8.24141 1.33738 8.07796L1.37204 8.03871L4.41042 5L1.37204 1.96129C1.20932 1.79858 1.20932 1.53476 1.37204 1.37204C1.52224 1.22184 1.75859 1.21028 1.92204 1.33738Z"
          />
        </svg>
      </span>
    </div>
  </div>
  <div
    class="tag-tag-0-2-1 tag-default-0-2-10 tag-bright-0-2-23"
    style="background: rgb(255, 255, 255); border-style: dashed;"
  >
    <span>
      + New Tag
    </span>
  </div>
</div>
`;

exports[`Tag[Input] should render correctly about input 1`] = `
<div>
  <div
    class="tag-tag-0-2-1 tag-default-0-2-10 tag-bright-0-2-23"
  >
    <div
      class="tag-inline-0-2-3"
    >
      <span>
        abc
      </span>
    </div>
    <div
      class="tag-closeIcon-0-2-7"
    >
      <span
        class="tag-closeIconWrapper-0-2-8"
      >
        <svg
          height="10"
          viewBox="0 0 10 10"
          width="10"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M1.92204 1.33738L1.96129 1.37204L5 4.41042L8.03871 1.37204C8.20142 1.20932 8.46524 1.20932 8.62796 1.37204C8.77816 1.52224 8.78972 1.75859 8.66262 1.92204L8.62796 1.96129L5.58958 5L8.62796 8.03871C8.79068 8.20142 8.79068 8.46524 8.62796 8.62796C8.47776 8.77816 8.24141 8.78972 8.07796 8.66262L8.03871 8.62796L5 5.58958L1.96129 8.62796C1.79858 8.79068 1.53476 8.79068 1.37204 8.62796C1.22184 8.47776 1.21028 8.24141 1.33738 8.07796L1.37204 8.03871L4.41042 5L1.37204 1.96129C1.20932 1.79858 1.20932 1.53476 1.37204 1.37204C1.52224 1.22184 1.75859 1.21028 1.92204 1.33738Z"
          />
        </svg>
      </span>
    </div>
  </div>
</div>
`;

exports[`Tag[Shape] should render correctly about shape 1`] = `
<div>
  <div
    class="tag-tag-0-2-1 tag-default-0-2-10 tag-bright-0-2-23"
  >
    <span>
      Default
    </span>
  </div>
  <div
    class="tag-tag-0-2-1 tag-rounded-0-2-26 tag-default-0-2-10 tag-bright-0-2-23"
  >
    <span>
      Rounded
    </span>
  </div>
</div>
`;
